### PR TITLE
Remove old Package Loader code and clean up repo

### PR DESCRIPTION
**Pull request by pramodya1994**
Fixes #<Issue Number>

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/util/RepoUtils.java [hunk 1]
```diff
@@ -18,23 +18,16 @@
 package org.wso2.ballerinalang.util;
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/util/RepoUtils.java:18]: Shall we remove this class as well? We can move the remaining utils to ProjectUtils and use that class.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java [hunk 2]
```diff
@@ -244,16 +240,13 @@ public SymbolEnter(CompilerContext context) {
...
-        this.sourceDirectory = context.get(SourceDirectory.class);
         this.importedPackages = new ArrayList<>();
         this.unknownTypeRefs = new HashSet<>();
         this.missingNodesHelper = BLangMissingNodesHelper.getInstance(context);
         this.packageCache = PackageCache.getInstance(context);
         this.intersectionTypes = new ArrayList<>();
 
         CompilerOptions options = CompilerOptions.getInstance(context);
-        projectAPIInitiatedCompilation = Boolean.parseBoolean(
-                options.get(CompilerOptionName.PROJECT_API_INITIATED_COMPILATION));
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java:256]: The compiler option name can be removed now. https://github.com/ballerina-platform/ballerina-lang/blob/1ee5bb5fc1655413c6d972d1ee57b67e76e68761/compiler/ballerina-lang/src/main/java/org/ballerinalang/compiler/CompilerOptionName.java#L68-L71

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/CentralUtils.java [hunk 3]
```diff
@@ -164,4 +168,30 @@ public static String getCentralPackageURL(String org, String pkgName) {
         }
         return BALLERINA_CENTRAL_PRODUCTION_URL + "/" + org + "/" + pkgName;
     }
+
+    /**
+     * Create and get the home repository path.
+     *
+     * @return home repository path
+     */
+    public static Path createAndGetHomeReposPath() {
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/CentralUtils.java:177]: Can't we use the `ProjectUtils.createAndGetHomeReposPath()`?

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java [hunk 4]
```diff
@@ -96,9 +96,10 @@
  * @since 2.0.0
  */
 public class ProjectUtils {
-    private static final String USER_HOME = "user.home";
+    public static final String USER_HOME = "user.home";
```
[the remark this answers is not in the document]
Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java:99]: That's what our usage of this at L450

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java (hunk 2 above)
Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java:256]: Added

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/CentralUtils.java (hunk 3 above)
Comment (pramodya1994) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/CentralUtils.java]: Done

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java]: Added to `ProjectConstants`

File: compiler/ballerina-lang/build.gradle [hunk 5]
```diff
@@ -51,6 +51,8 @@ checkstyleMain {
 }
 
 test {
+    systemProperty "BOOTSTRAP_LANG_LIB", "false"
+
```
Comment (pramodya1994) [inline compiler/ballerina-lang/build.gradle:55]:
```suggestion
```

**Comment by pramodya1994** [comment]
Added a new PR https://github.com/ballerina-platform/ballerina-lang/pull/31073

**Pull request closed without merging**